addons/mail/models/mail_activity.py

```diff
@@ -757,6 +760,23 @@ def _search_activity_type_id(self, operator, operand):
     def _search_activity_summary(self, operator, operand):
         return [('activity_ids.summary', operator, operand)]
 
+    @api.depends('activity_ids.date_deadline', 'activity_ids.user_id')
+    @api.depends_context('uid')
+    def _compute_my_activity_date_deadline(self):
+        for record in self:
+            record.my_activity_date_deadline = next((
+                activity.date_deadline
+                for activity in record.activity_ids
+                if activity.user_id.id == record.env.uid
+            ), False)
+
+    def _search_my_activity_date_deadline(self, operator, operand):
+        activity_ids = self.env['mail.activity']._search([
```

Comment:
We should also include a domain part about model being matching self, to avoid searching activities linked to other models.

Reply:
@tde-banana-odoo Done!!